main.go

```diff
@@ -243,6 +227,34 @@ func StartPurgeLoop(wg *sync.WaitGroup, ctx context.Context, secInterval int, ch
 	}
 }
 
+func PreprocessAnalyticsValues(AnalyticsValues []interface{}, serializerMethod serializer.AnalyticsSerializer, analyticsKeyName string, omitDetails bool, job *health.Job, startTime time.Time, secInterval int) {
+	keys := make([]interface{}, len(AnalyticsValues))
+
+	for i, v := range AnalyticsValues {
+		decoded := analytics.AnalyticsRecord{}
+		err := serializerMethod.Decode([]byte(v.(string)), &decoded)
+
+		log.WithFields(logrus.Fields{
+			"prefix": mainPrefix,
+		}).Debug("Decoded Record: ", decoded)
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"prefix":       mainPrefix,
+				"analytic_key": analyticsKeyName,
+			}).Error("Couldn't unmarshal analytics data:", err)
```

Comment:
Perhaps just add `continue` after line 244 and remove the block on 245.

Reply:
done